### ISSUE TITLE
Kani module stubs for development

### ISSUE THREAD

**Issue by ajrudzitis** [model-checking/kani#3778]
Requested feature: Kani module stubs for development
Use case:

Kani is currently difficult to use in an IDE because the kani module is only injected when running `cargo kani`. This means that while writing harnesses in the IDE along with my code, the IDE does not recognize any statements like `kani::any()`. This adds friction to writing new test harnesses and I cannot use facilities in the IDE to identify issues before I run my harness. 

If there could a module published that provided stubs for the Kani API, it would be useful for allowing the IDE to do basic type checks, etc.

**Comment by celinval** [comment]
Yes, this is definitely something that's been in our radar. I think this would also play nicely with things like #2035.

I think the main question is how to ensure compatibility between the crate version a user is importing and the one Kani is using. Especially if dependencies start exporting things like `Arbitrary` implementation.

**Comment by zhassan-aws** [comment]
Thanks, this is very helpful! We'll add these instructions to the documentation (or feel free to open a PR yourself to do that).
